pkg/network/ovn_kubernetes.go

```diff
@@ -154,6 +154,12 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 		}
 	}
 
+	if len(bootstrapResult.OVN.MasterIPs) == 1 {
```

Comment:
looks good, leverages the existing field in the bootstrap (https://github.com/openshift/cluster-network-operator/blob/7498fab6bdc2b872c2f09e9a35eb19794a4c0fd7/pkg/bootstrap/types.go#L37)

Q: Does single master = SNO always? Aren't the number of master nodes in OpenShift configurable in the sense, maybe we have multiple nodes but single master? I am not sure, hence asking. Is this the standard way of recognizing if a cluster is SNO or should we actually check if there is only one node in the cluster?

Reply:
Good question. My understanding was that Openshift has >= 3 masters and just in case of SNO only one. But I am not sure as you're asking. Asked it on [slack](https://coreos.slack.com/archives/C018KQE33MF/p1627304315423500) Any idea who can answer this as well?

Reply:
@romfreiman: fyi

Reply:
Also on second thoughts, why do we do leader election in SNO? Shouldn't we just remove it all together?

Reply:
@tssurya yes we should :) But we dont know how every component behaves without leader and we cannot enforce it at this point, so we dont provide such guideline. If you can/know how to remove it, feel free. IMO it's even better.

cc: @chaitanyaenr

Reply:
@romfreiman : ack yea I understand. Not sure about how other components do it, but we can see if it can be removed in ovn-k for SNOs.

Reply:
@tssurya sounds great! btw, we have a sno weekly today if you would like to participate.

Reply:
have a pr for this open in ovn-kubernetes: https://github.com/ovn-org/ovn-kubernetes/pull/2364

Reply:
as pointed out by @aojea in [slack](https://coreos.slack.com/archives/CDCP2LA9L/p1627412484098900?thread_ts=1627402405.090600&cid=CDCP2LA9L) there can be scale out cases and then not having leader election will create a mess.

Reply:
After the discussion on [slack](https://coreos.slack.com/archives/CDCP2LA9L/p1627402405090600), we will increase the timeouts. I will add a code comment with a link to the slack discussion